doc/mgmt/config_replace_rollback_HLD.md

```diff
@@ -0,0 +1,415 @@
+- A new CLI (replace) command to perform configuration replace from configure session mode.
+- A "timer" option added to commit command for user to apply the change to CONFIG_DB on trial basis, if a timer gets expired, rollback to the saved backup file.
+- A "confirm" option added to commit command for user to confirm new configuration changes, i.e. move trial to permanent stage.
+- When "Kill" is executed
+	+ timer will be canceled and running-configuration reverted back to previous configuration state.
+
+**Non configure session CLI**
+- A "replace" option added to existing CLI "copy source-url running-configuration" command to perform configuration replacement on exec mode.
+
+**REST/gNMI/gNOI** 
+- A new REST POST operation to perform config replacement.
+- A new gNMI/gNOI operation to perform config replacement.
+
+
+## 1.3 Design Overview
+### 1.3.1 Basic Approach
+The config replace design will leverage the following configure session functionality as well as extend it as specified.
+
+- When a configure session is opened, CONFIG_DB is locked and an unique session token is assigned to the requests from config session to differentiate from non-session based requests. DB writes will be rejected if requests are initiated from REST, gNMI/gNOI including non-configure session CLI.
+- A session token is used for creating, viewing checkpoints optionally with label. Non-session based operations will not be using such.
+- Commit command for cached candidate datastore is non-modeled CLI-only RPCs. Extension required for configure replace use, e.g. commit timer, commit confirm
+- All the changes within a session will be cached to candidate datastore, not to CONFIG_DB until a commit (with or without timer) is issued. Config replace from non configure session mode is to apply changes directly to CONFIG_DB.
+
+
+When a configure replace resquest is sent from non-configure session or REST, gNMI/gNOI:
+- Check if CONFIG_DB is locked. If it is locked, the command is rejected to avoid conflicting configuration state.
+
+When a configure replace request is sent within a configure session:
+- When 'replace' command is issued, check if the candidate store is dirtified. If it is dirtified, the command is rejected to avoid conflicting configuration state.
+- Since DB writes are targeted to candidate datastore, CLI 'replace' command is followed by commit with/wthout <timer> to apply changes to CONFIG_DB.
+- "commit" command without a timer saves the changes to CONFIG_DB. Chekcpoints will be generated by configure session.
+- "commit <timer>" command saves the changes to CONFIG_DB and put in trial stage until it is confirmed. A temporarty backup file for current config will be saved for rollback. if a timer gets expired without confirm, it will be rolled back to the previous configuration state by internal config replacement with saved backup file. If an error occurred during rollback, it may require manual troubleshooting to fix the problem state. 
```

Comment:
"<timer>" is not seen in the md file output. It only says "commit "

Reply:
Backslash('\') need to be used as escape character. 
`- "commit \<timer\>"`
We will add it